shell/common/shell_test_platform_view.cc

```diff
@@ -32,28 +21,18 @@ std::unique_ptr<ShellTestPlatformView> ShellTestPlatformView::Create(
   // Make this fully runtime configurable
   switch (backend) {
     case BackendType::kDefaultBackend:
-#ifdef SHELL_ENABLE_GL
     case BackendType::kGLBackend:
-      return std::make_unique<ShellTestPlatformViewGL>(
+      return CreatePlatformViewGL(
           delegate, task_runners, vsync_clock, create_vsync_waiter,
-          shell_test_external_view_embedder);
-#endif  // SHELL_ENABLE_GL
-#ifdef SHELL_ENABLE_VULKAN
+          shell_test_external_view_embedder, is_gpu_disabled_sync_switch);
     case BackendType::kVulkanBackend:
-      return std::make_unique<ShellTestPlatformViewVulkan>(
+      return CreatePlatformViewVulkan(
           delegate, task_runners, vsync_clock, create_vsync_waiter,
-          shell_test_external_view_embedder);
-#endif  // SHELL_ENABLE_VULKAN
-#ifdef SHELL_ENABLE_METAL
+          shell_test_external_view_embedder, is_gpu_disabled_sync_switch);
     case BackendType::kMetalBackend:
-      return std::make_unique<ShellTestPlatformViewMetal>(
+      return CreatePlatformViewMetal(
           delegate, task_runners, vsync_clock, create_vsync_waiter,
           shell_test_external_view_embedder, is_gpu_disabled_sync_switch);
```

Comment:
Instead of instantiating all the backend-specific stuff here, the implementations are now in the appropriate `shell_test_platform_view_metal.mm` or `shell_test_platform_view_vulkan.cc` etc. file, with fallback implementations in this file for any backends that are disabled.